coderd/parameters.go

```diff
@@ -269,3 +275,31 @@ func (api *API) getWorkspaceOwnerData(
 		Groups:       groupNames,
 	}, nil
 }
+
+// parameterProvisionerVersionDiagnostic checks the version of the provisioner
+// used to create the template version. If the version is less than 1.5, it
+// returns a warning diagnostic. Only versions 1.5+ return the module & plan data
+// required.
+func parameterProvisionerVersionDiagnostic(tf database.TemplateVersionTerraformValue) hcl.Diagnostics {
+	missingMetaData := hcl.Diagnostic{
+		Severity: hcl.DiagError,
+		Summary:  "This template version is missing required metadata to support dynamic parameters. Go back to the classic creation flow.",
```

Comment:
hmm. I don't like this. I would much prefer if we could have a way to expose this naturally to the frontend, forcing it over to the classic form. why make everyone click the "dynamic parameters suck" button? that's a super bad first impression

Reply:
You are right, it is unfortunate. I want to get this in first, just to get an error up if someone toggles on dev and get this detection in. I will build on this experience.

We just need _something_ now, otherwise it fails silently.